packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -1624,27 +1656,28 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           const node: N.CallExpression = this.startNodeAt(startPos, startLoc);
           node.callee = base;
 
-          const typeArguments = this.tsParseTypeArguments();
-
-          if (typeArguments) {
-            if (!noCalls && this.eat(tt.parenL)) {
-              // possibleAsync always false here, because we would have handled it above.
-              // $FlowIgnore (won't be any undefined arguments)
-              node.arguments = this.parseCallExpressionArguments(
-                tt.parenR,
-                /* possibleAsync */ false,
-              );
-              node.typeParameters = typeArguments;
-              return this.finishCallExpression(node);
-            } else if (this.match(tt.backQuote)) {
-              return this.parseTaggedTemplateExpression(
-                startPos,
-                startLoc,
-                base,
-                state,
-                typeArguments,
-              );
-            }
+          const typeArguments = this.tsTryParseTypeArguments();
```

Comment:
Wouldn't this add an additional try catch?

Reply:
Ok my mistake, I should have looked first what `tsTryParseDelimitedList` is doing. I assumed everything with a try in the name uses a tray catch.

Reply:
Maybe we should rename things which don't use `try/catch` to `tsMaybeParse*`